Build failures with LTO and `-Werror=odr,lto-type-mismatch`

# Bug Report

**Current behavior**
The software cannot be built with LTO enabled and appropriate error flags enabled: `-Werror=odr,lto-type-mismatch`

**Expected behavior**
The software should compile successfully with these errors enabled.

**Steps to reproduce:**
Build with the above options, or:

`-march=native -O2 -pipe -flto -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing`

**Gridcoin version**

5.4.9.0

**Machine specs**
- OS: Gentoo Linux
- CPU: AMD Ryzen 9 5950X
- RAM: 128G
- Disk size: 1 + 2 TB
- Disk Type (HD/SSD): NVMe

**Extra information**

Compiler: GCC 15.2.0
build log: [gridcoin-5.4.9.0-r1:20251227-234617.log.gz](https://github.com/user-attachments/files/24357287/gridcoin-5.4.9.0-r1.20251227-234617.log.gz)


Issues:

```
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0_build/src/qt/gridcoinqt_autogen/EWIEGA46WW/../../../../../Gridcoin-Research-5.4.9.0/src/qt/notificator.h:15:7: error: type ‘struct Notificator’ violates the C++ One Definition Rule [-Werror=odr]
   15 | class Notificator: public QObject
      |       ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.h:15:7: note: a different type is defined in another translation unit
   15 | class Notificator: public QObject
      |       ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0_build/src/qt/gridcoinqt_autogen/EWIEGA46WW/../../../../../Gridcoin-Research-5.4.9.0/src/qt/notificator.h:58:21: note: the first difference of corresponding definitions is field ‘interface’
   58 |     QDBusInterface *interface;
      |                     ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.h:15:7: note: a type with different number of fields is defined in another translation unit
   15 | class Notificator: public QObject
      |                  ^
```

and 

```
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.h:22:5: error: type of ‘__ct_comp ’ does not match original declaration [-Werror=lto-type-mismatch]
   22 |     Notificator(const QString &programName, QSystemTrayIcon *trayIcon, QWidget *parent);
      |     ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.cpp:35:1: note: ‘__ct_comp ’ was previously declared here
   35 | Notificator::Notificator(const QString &_programName, QSystemTrayIcon *_trayIcon, QWidget *_parent) :
      | ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.cpp:35:1: note: code may be misoptimized unless ‘-fno-strict-aliasing’ is used
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0_build/src/qt/gridcoinqt_autogen/EWIEGA46WW/../../../../../Gridcoin-Research-5.4.9.0/src/qt/notificator.h:43:10: error: type of ‘notify’ does not match original declaration [-Werror=lto-type-mismatch]
   43 |     void notify(Class cls, const QString &title, const QString &text,
      |          ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.h:43:10: error: type of ‘notify’ does not match original declaration [-Werror=lto-type-mismatch]
   43 |     void notify(Class cls, const QString &title, const QString &text,
      |          ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.cpp:233:6: note: ‘notify’ was previously declared here
  233 | void Notificator::notify(Class cls, const QString &title, const QString &text, const QIcon &icon, int millisTimeout)
      |      ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.cpp:233:6: note: code may be misoptimized unless ‘-fno-strict-aliasing’ is used
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0_build/src/qt/gridcoinqt_autogen/EWIEGA46WW/../../../../../Gridcoin-Research-5.4.9.0/src/qt/notificator.h:23:5: error: type of ‘__dt_del ’ does not match original declaration [-Werror=lto-type-mismatch]
   23 |     ~Notificator();
      |     ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.cpp:65:1: note: ‘__dt_del ’ was previously declared here
   65 | Notificator::~Notificator()
      | ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.cpp:65:1: note: code may be misoptimized unless ‘-fno-strict-aliasing’ is used
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0_build/src/qt/gridcoinqt_autogen/EWIEGA46WW/../../../../../Gridcoin-Research-5.4.9.0/src/qt/notificator.h:23:5: error: type of ‘__dt_comp ’ does not match original declaration [-Werror=lto-type-mismatch]
   23 |     ~Notificator();
      |     ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.cpp:65:1: note: ‘__dt_comp ’ was previously declared here
   65 | Notificator::~Notificator()
      | ^
/var/tmp/portage/net-p2p/gridcoin-5.4.9.0-r1/work/Gridcoin-Research-5.4.9.0/src/qt/notificator.cpp:65:1: note: code may be misoptimized unless ‘-fno-strict-aliasing’ is used
lto1: some warnings being treated as errors
```

--- a/src/qt/notificator.cpp
+++ b/src/qt/notificator.cpp
@@ -37,10 +37,8 @@ QObject(_parent),
 parent(_parent),
 programName(_programName),
 mode(None),
-trayIcon(_trayIcon)
-#ifdef USE_DBUS
-    ,interface(nullptr)
-#endif
+trayIcon(_trayIcon),
+interface(nullptr)
 {
     if(_trayIcon && _trayIcon->supportsMessages())
     {

--- a/src/qt/notificator.h
+++ b/src/qt/notificator.h
@@ -6,9 +6,7 @@
 
 QT_BEGIN_NAMESPACE
 class QSystemTrayIcon;
-#ifdef USE_DBUS
 class QDBusInterface;
-#endif
 QT_END_NAMESPACE
 
 /** Cross-platform desktop notification client. */
@@ -54,11 +52,9 @@ private:
     QString programName;
     Mode mode;
     QSystemTrayIcon *trayIcon;
-#ifdef USE_DBUS
     QDBusInterface *interface;
 
     void notifyDBus(Class cls, const QString &title, const QString &text, const QIcon &icon, int millisTimeout);
-#endif
     void notifySystray(Class cls, const QString &title, const QString &text, int millisTimeout);
 #ifdef Q_OS_MAC
     void notifyMacUserNotificationCenter(const QString &title, const QString &text);

